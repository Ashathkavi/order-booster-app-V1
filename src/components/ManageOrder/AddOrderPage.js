import React from 'react'
import {OrderForm} from './OrderForm'
import {startAddOrder} from '../../actions/orders'
import {connect} from 'react-redux'


export const AddOrderPage = (props) => {

    
    const counter = (orders) =>{
        let count = orders.length + 1
        let i = true
        while(i === true){
            orders.map((order) => {
                i=false
                if(order.count  === count){
                    count += 1
                    i=true
                }
            })
        }
        return count
    } 

    const onSubmit = (order) => {
        props.startAddOrder({...order, count:counter(props.orders)})
        props.history.push('/order')
    }

    return(
        <div>
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Add Order</h1>
                </div>
            </div>

            <div className="content-container">
                <OrderForm onSubmit={ onSubmit}/>
            </div>
            

        </div>
)}


const mapDispatchToProps = (dispatch) => ({
    startAddOrder: (order) => dispatch(startAddOrder(order))
})


const mapStateToProps = (state) => {
    console.log(state.orders)
    return {
        orders:state.orders
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddOrderPage)