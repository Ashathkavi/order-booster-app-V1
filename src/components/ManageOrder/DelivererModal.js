import React, {useEffect, useState} from 'react'
import Modal from 'react-modal'
import CategoryList from '../../fixtures/FoodCategoryList'
import database from '../../firebase/firebase'


class Deliverer{
    constructor(name, uid){
        this.name = name
        this.uid = uid
    }
}



const DelivererModal = (props) => {

    const [deliverers, setDeliverers] = useState([])

    useEffect(()=>{
        const deliverers = []
        database.ref(`users`).once('value')
            .then((snap)=>{
                snap.forEach((childSnapshot)=>{
                    if(childSnapshot.val().role === 'deliverer'){
                        const deliverer = new Deliverer(childSnapshot.val().name, childSnapshot.key)
                        deliverers.push(deliverer)                        
                    }
                })
                setDeliverers(deliverers)                    
            })
            .catch((error)=>console.log('User gaining request failed :', error))
        
    })

    return(
        <Modal
            isOpen={props.modalIsOpen}
            contentLabel = "Please Select a Deliverer"
            onRequestClose = {props.handleCloseModal}
        >
            <h3>Please Select a Deliverer</h3>
            {
                deliverers.map((deliverer)=>(
                    <button onClick={() =>{                
                        props.onDelivery(deliverer.uid)
                        props.handleCloseModal()
                    }}>{deliverer.name}</button>
                ))
            }
            
        </Modal>
)}

export default DelivererModal;


/*

*/