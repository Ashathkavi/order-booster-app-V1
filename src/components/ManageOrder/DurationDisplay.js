import React, {useState,useEffect} from 'react'
import moment from 'moment'


const DurationDisplay = ({orderEndTime}) => {
    const [time, setTime] = useState(moment().valueOf())

    useEffect(() => {
        setTimeout(() => setTime(moment().valueOf()), 60000)
    }, [time])

    let duration = moment(orderEndTime).diff(moment())
    let durationPlusTen = moment(orderEndTime).subtract(10,'minutes').diff(moment())
    if(duration < 0){
        if(moment.utc(duration*-1).format('DD')==='01'){
            return <span style={{color: "red"}}>{moment.utc(duration*-1).format('HH ')}h&nbsp;&nbsp;{moment.utc(duration*-1).format('mm ')}min</span>
        }else{
            return <span style={{color: "red"}}>
                        {moment.utc(duration*-1).format('DD ')}d&nbsp;&nbsp;
                        {moment.utc(duration*-1).format('HH ')}h&nbsp;&nbsp;
                        {moment.utc(duration*-1).format('mm ')}min
                    </span>
        }
        //console.log(moment.utc(duration*-1).format('DD'))
    }else if(durationPlusTen < 0){
        return <span style={{color: "orange"}}>{moment.utc(duration).format('mm ') }min</span>
    }else{
        if(moment.utc(duration).format('HH')==='00'){
            return <span >{moment.utc(duration).format('mm ')}min</span>
        }else{
            return <span >{moment.utc(duration).format('HH ')}h&nbsp;&nbsp;{moment.utc(duration).format('mm ')}min</span>
        }
        
    }
    

} 

export default DurationDisplay