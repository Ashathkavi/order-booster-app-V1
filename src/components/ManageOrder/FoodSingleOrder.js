import React from 'react'
import {Link} from 'react-router-dom'

const FoodSingleOrder = (props) => (
    <tr>
        <td>{props.singleTypeOrder.food.name}</td>
        <td>{props.singleTypeOrder.foodQuantity}</td>
        <td>{props.singleTypeOrder.prepared ? "true" : "false"}</td>
        {!!props.onRemoveSingleOrder && <td><button type='button' onClick={()=>props.onRemoveSingleOrder(props.iterable)}>Remove</button></td>}
            
    </tr>
)

export default FoodSingleOrder