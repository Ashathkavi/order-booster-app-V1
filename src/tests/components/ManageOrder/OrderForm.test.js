import React from 'react'
import {shallow} from 'enzyme'
import {OrderForm} from '../../../components/ManageOrder/OrderForm'
import sampleOrders from '../../../fixtures/sampleOrders'
import moment from 'moment'
import DatePicker from 'react-datepicker'
import {OrderModal} from '../../../components/ManageOrder/OrderModal'
import FoodSingleOrder from '../../../components/ManageOrder/FoodSingleOrder'


const sample_order = sampleOrders()
const now = moment().valueOf()

test('should render OrderForm correctly', () => {
    const wrapper = shallow(<OrderForm/>)
    expect(wrapper).toMatchSnapshot()
})

test('should render OrderForm correctly with order date', () => {
    const wrapper = shallow(<OrderForm order= {sample_order[0]}/>)
    expect(wrapper).toMatchSnapshot()
})



test('should render error for invalid form submission', () => {
    const wrapper = shallow(<OrderForm/>)
    expect(wrapper).toMatchSnapshot()
    wrapper.find('form').simulate('submit', {
        preventDefault: () => { }
    })
    expect(wrapper.state('error').length).toBeGreaterThan(0)
    expect(wrapper).toMatchSnapshot()
})



test('should set customer name on input change', () => {
    const value = 'Divasthan'
    const wrapper = shallow(<OrderForm/>)
    wrapper.find('input').at(0).simulate('change', {
        target:{value},
        persist: () => { }
    })
    expect(wrapper.state('customerName')).toBe(value)
})

test('should set phoneNumber on input change', () => {
    const value = '0762352885'
    const wrapper = shallow(<OrderForm/>)
    wrapper.find('input').at(1).simulate('change', {
        target:{value},
        persist: () => { }
    })
    expect(wrapper.state('phoneNumber')).toBe(value)
})

test('should set address on input change', () => {
    const value = 'kallady'
    const wrapper = shallow(<OrderForm/>)
    wrapper.find('input').at(2).simulate('change', {
        target:{value},
        persist: () => { }
    })
    expect(wrapper.state('address')).toBe(value)
})

test('should set description on input change', () => {
    const value = 'less spicy'
    const wrapper = shallow(<OrderForm/>)
    wrapper.find('textarea').at(0).simulate('change', {
        target:{value},
        persist: () => { }
    })
    expect(wrapper.state('description')).toBe(value)
})

test('should set status on input change', () => {
    const value = 'confirmed'
    const wrapper = shallow(<OrderForm/>)
    wrapper.find('select').at(0).simulate('change', {
        target:{value},
        persist: () => { }
    })
    expect(wrapper.state('status')).toBe(value)
    expect(wrapper.state('statusTime')).toBe(moment().valueOf())
})

test('should set kot status on input change', () => {
    const value = 'passed'
    const wrapper = shallow(<OrderForm/>)
    wrapper.find('select').at(1).simulate('change', {
        target:{value},
        persist: () => { }
    })
    expect(wrapper.state('kotStatus')).toBe(value)
    expect(wrapper.state('kotStatusTime')).toBe(moment().valueOf())
})

test('should set bill status on input change', () => {
    const value = 'printed'
    const wrapper = shallow(<OrderForm/>)
    wrapper.find('select').at(2).simulate('change', {
        target:{value},
        persist: () => { }
    })
    expect(wrapper.state('billStatus')).toBe(value)
    expect(wrapper.state('billStatusTime')).toBe(moment().valueOf())
})

test('should set duration on input change', () => {
    const value = '20'
    const wrapper = shallow(<OrderForm/>)
    wrapper.find('select').at(3).simulate('change', {
        target:{value},
        persist: () => { }
    })
    const orderEndTime = moment().add(value,'minutes').valueOf()
    expect(wrapper.state('orderEndTime')).toBe(orderEndTime)
})


test('should call onSubmit props for valid form submission', () => {
    let billAmount = 0
    sample_order[0].foods.map((food)=>billAmount=(food.food.amount*food.foodQuantity )+ billAmount)
    
    const onSubmitSpy = jest.fn()
    const wrapper = shallow(<OrderForm order={sample_order[0]} onSubmit={onSubmitSpy}/>)
    wrapper.find('form').simulate('submit', {
        preventDefault:() => {}
    })
    expect(wrapper.state('error')).toBe('')
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        address:sample_order[0].address,
        amount:billAmount,
        createdAt:sample_order[0].createdAt,
        billStatus:sample_order[0].billStatus,
        customerName:sample_order[0].customerName,
        description:sample_order[0].description,
        foods:sample_order[0].foods,
        kotStatus:sample_order[0].kotStatus, 
        orderEndTime:sample_order[0].orderEndTime,
        phoneNumber:sample_order[0].phoneNumber,
        status:sample_order[0].status 
    
    })
})

test('should set new date on date change', () => {
    const now = moment()
    const wrapper = shallow(<OrderForm/>)
    wrapper.find(DatePicker).prop('onChange')(now.toDate())
    expect(wrapper.state('createdAt')).toEqual(now.valueOf())

})


test('should toggle the modal visibility  when click the add food buttton', () => {
    const wrapper = shallow(<OrderForm/>)
    //console.log(wrapper.state('isModalOpen'))
    wrapper.find('button').at(0).simulate('click')
    expect(wrapper.state('isModalOpen')).toEqual(true)

})

test('should toggle the modal visibility  when close the modal ', () => {
    const wrapper = shallow(<OrderForm/>)
    //console.log(wrapper.state('isModalOpen'))
    wrapper.find('button').at(0).simulate('click')
    //console.log(wrapper.state('isModalOpen'))
    wrapper.find(OrderModal).prop('onVisibleChange')()
    expect(wrapper.state('isModalOpen')).toEqual(false)

})


test('should set new foods on food selection', () => {
    const newSingleOrder = sample_order[0].foods
    const wrapper = shallow(<OrderForm/>)
    wrapper.find(OrderModal).prop('onSingleOrderChange')(newSingleOrder)
    expect(wrapper.state('foods')).toEqual(newSingleOrder)

})


test('should remove 1st singleorder food', () => {
    const positionOfSingleOrder = 0
    const newSingleOrder = sample_order[0].foods
    const wrapper = shallow(<OrderForm/>)
    wrapper.find(OrderModal).prop('onSingleOrderChange')(newSingleOrder);
    wrapper.find(FoodSingleOrder).at(positionOfSingleOrder).prop('onRemoveSingleOrder')(positionOfSingleOrder)
    expect(wrapper.state('foods')).toEqual([newSingleOrder[1]])

})

test('should remove 2nd singleorder food', () => {
    const positionOfSingleOrder = 1
    const newSingleOrder = sample_order[0].foods
    const wrapper = shallow(<OrderForm/>)
    wrapper.find(OrderModal).prop('onSingleOrderChange')(newSingleOrder);
    wrapper.find(FoodSingleOrder).at(positionOfSingleOrder).prop('onRemoveSingleOrder')(positionOfSingleOrder)
    expect(wrapper.state('foods')).toEqual([newSingleOrder[0]])

})

test(' shoul calculate bill amount corectly',()=>{
    
})

