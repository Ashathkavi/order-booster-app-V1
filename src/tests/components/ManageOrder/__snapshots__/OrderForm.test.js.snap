// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render OrderForm correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Add Food
    </button>
    <Connect(OrderModal)
      isModalOpen={false}
      onSingleOrderChange={[Function]}
      onVisibleChange={[Function]}
      singleTypeOrders={Array []}
    />
    <input
      onChange={[Function]}
      placeholder="Customer Name"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Phone Number"
      type="number"
      value={0}
    />
    <input
      onChange={[Function]}
      placeholder="Customer Address"
      type="text"
      value=""
    />
    <textarea
      autoFocus={true}
      onChange={[Function]}
      placeholder="Add a discription for your order (optional)"
      value=""
    />
    <div>
      <h3>
        :::::::::::::
        0
        :::::::::::::
      </h3>
    </div>
    <select
      onChange={[Function]}
      value="confirmed"
    >
      <option
        value="confirmed"
      >
        Confirmed
      </option>
      <option
        value="cancelled"
      >
        Cancelled
      </option>
      <option
        value="kitchen"
      >
        Kitchen
      </option>
      <option
        value="table"
      >
        Table
      </option>
      <option
        value="on delivery"
      >
        On Delivery
      </option>
      <option
        value="recieved"
      >
        Recieved
      </option>
    </select>
    <select
      onChange={[Function]}
      value=""
    >
      <option
        selected={true}
        value=""
      >
        Select a Deliverer
      </option>
      <option
        value="defaultId"
      >
        default
      </option>
    </select>
    <select
      disabled={true}
      onChange={[Function]}
      value="not"
    >
      <option
        value="not"
      >
        Not Passed
      </option>
      <option
        value="passed"
      >
        Passed
      </option>
    </select>
    <select
      disabled={true}
      onChange={[Function]}
      value="not"
    >
      <option
        value="not"
      >
        Not Printed
      </option>
      <option
        value="printed"
      >
        Printed
      </option>
    </select>
    <select
      onChange={[Function]}
      value="20"
    >
      <option
        value="20"
      >
        20 minutes
      </option>
      <option
        value="30"
      >
        30 minutes
      </option>
      <option
        value="40"
      >
        40 minutes
      </option>
      <option
        value="50"
      >
        50 minutes
      </option>
      <option
        value="60"
      >
        1 hour
      </option>
      <option
        value="70"
      >
        1 hour 10 minutes
      </option>
      <option
        value="80"
      >
        1 hour 20 minute
      </option>
      <option
        value="90"
      >
        1 and 1\\2 hour 
      </option>
    </select>
    <r
      allowSameDay={false}
      calendarStartDay={0}
      customTimeInput={null}
      dateFormat="MMMM d, yyyy h:mm aa"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      enableTabLoop={true}
      excludeScrollbar={true}
      focusSelectedMonth={false}
      monthsShown={1}
      nextMonthButtonLabel="Next Month"
      nextYearButtonLabel="Next Year"
      onBlur={[Function]}
      onCalendarClose={[Function]}
      onCalendarOpen={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      preventOpenOnFocus={false}
      previousMonthButtonLabel="Previous Month"
      previousYearButtonLabel="Previous Year"
      readOnly={false}
      renderDayContents={[Function]}
      selected={1970-01-01T00:00:00.000Z}
      shouldCloseOnSelect={true}
      showFourColumnMonthYearPicker={false}
      showFullMonthYearPicker={false}
      showMonthYearPicker={false}
      showPopperArrow={true}
      showPreviousMonths={false}
      showQuarterYearPicker={false}
      showTimeInput={false}
      showTimeSelect={true}
      showTwoColumnMonthYearPicker={false}
      showYearPicker={false}
      strictParsing={false}
      timeCaption="Time"
      timeFormat="HH:mm"
      timeInputLabel="Time"
      timeIntervals={30}
      withPortal={false}
      yearItemNumber={12}
    />
    <button>
      Save Order
    </button>
  </form>
  <div>
    <p>
      customerName: 
    </p>
    <p>
      phoneNumber: 
      0
    </p>
    <p>
      address: 
    </p>
    <p>
      description: 
    </p>
    <p>
      orderEndTime:  
      Thu Jan 01 1970 05:50:00 GMT+0530
    </p>
    <p>
      status: 
      confirmed
    </p>
    <p>
      kotSatus: 
      not
    </p>
    <p>
      billStatus: 
      not
    </p>
    <p>
      createdAt: 
      Thu Jan 01 1970 05:30:00 GMT+0530
    </p>
    <p>
      foods: 
    </p>
    <p>
      isModaOpen: 
      false
    </p>
    <p>
      deliverers: 
    </p>
  </div>
</div>
`;

exports[`should render OrderForm correctly with order date 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Add Food
    </button>
    <Connect(OrderModal)
      isModalOpen={false}
      onSingleOrderChange={[Function]}
      onVisibleChange={[Function]}
      singleTypeOrders={
        Array [
          SingleOrder {
            "food": Object {
              "amount": 380,
              "category": "Fried Rice",
              "createdAt": 100001,
              "description": "",
              "foodSize": "regular",
              "id": "2",
              "largeAvailability": true,
              "name": "Beef Fried Rice",
            },
            "foodQuantity": 2,
          },
          SingleOrder {
            "food": Object {
              "amount": 350,
              "category": "Fried Rice",
              "createdAt": 100000,
              "description": "",
              "foodSize": "regular",
              "id": "1",
              "largeAvailability": true,
              "name": "Chicken Fried Rice",
            },
            "foodQuantity": 3,
          },
        ]
      }
    />
    <input
      onChange={[Function]}
      placeholder="Customer Name"
      type="text"
      value="Ashath"
    />
    <input
      onChange={[Function]}
      placeholder="Phone Number"
      type="number"
      value="0765822562"
    />
    <input
      onChange={[Function]}
      placeholder="Customer Address"
      type="text"
      value="Mamangam"
    />
    <textarea
      autoFocus={true}
      onChange={[Function]}
      placeholder="Add a discription for your order (optional)"
      value="less oil"
    />
    <FoodSingleOrder
      iterable={0}
      key="0"
      onRemoveSingleOrder={[Function]}
      singleTypeOrder={
        SingleOrder {
          "food": Object {
            "amount": 380,
            "category": "Fried Rice",
            "createdAt": 100001,
            "description": "",
            "foodSize": "regular",
            "id": "2",
            "largeAvailability": true,
            "name": "Beef Fried Rice",
          },
          "foodQuantity": 2,
        }
      }
    />
    <FoodSingleOrder
      iterable={1}
      key="1"
      onRemoveSingleOrder={[Function]}
      singleTypeOrder={
        SingleOrder {
          "food": Object {
            "amount": 350,
            "category": "Fried Rice",
            "createdAt": 100000,
            "description": "",
            "foodSize": "regular",
            "id": "1",
            "largeAvailability": true,
            "name": "Chicken Fried Rice",
          },
          "foodQuantity": 3,
        }
      }
    />
    <div>
      <h3>
        :::::::::::::
        1810
        :::::::::::::
      </h3>
    </div>
    <select
      onChange={[Function]}
      value="confirmed"
    >
      <option
        value="confirmed"
      >
        Confirmed
      </option>
      <option
        value="cancelled"
      >
        Cancelled
      </option>
      <option
        value="kitchen"
      >
        Kitchen
      </option>
      <option
        value="table"
      >
        Table
      </option>
      <option
        value="on delivery"
      >
        On Delivery
      </option>
      <option
        value="recieved"
      >
        Recieved
      </option>
    </select>
    <select
      onChange={[Function]}
      value="Ashath"
    >
      <option
        selected={true}
        value=""
      >
        Select a Deliverer
      </option>
      <option
        value="defaultId"
      >
        default
      </option>
    </select>
    <select
      disabled={false}
      onChange={[Function]}
      value="passed"
    >
      <option
        value="not"
      >
        Not Passed
      </option>
      <option
        value="passed"
      >
        Passed
      </option>
    </select>
    <select
      disabled={false}
      onChange={[Function]}
      value="printed"
    >
      <option
        value="not"
      >
        Not Printed
      </option>
      <option
        value="printed"
      >
        Printed
      </option>
    </select>
    <select
      onChange={[Function]}
      value="20"
    >
      <option
        value="20"
      >
        20 minutes
      </option>
      <option
        value="30"
      >
        30 minutes
      </option>
      <option
        value="40"
      >
        40 minutes
      </option>
      <option
        value="50"
      >
        50 minutes
      </option>
      <option
        value="60"
      >
        1 hour
      </option>
      <option
        value="70"
      >
        1 hour 10 minutes
      </option>
      <option
        value="80"
      >
        1 hour 20 minute
      </option>
      <option
        value="90"
      >
        1 and 1\\2 hour 
      </option>
    </select>
    <r
      allowSameDay={false}
      calendarStartDay={0}
      customTimeInput={null}
      dateFormat="MMMM d, yyyy h:mm aa"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      enableTabLoop={true}
      excludeScrollbar={true}
      focusSelectedMonth={false}
      monthsShown={1}
      nextMonthButtonLabel="Next Month"
      nextYearButtonLabel="Next Year"
      onBlur={[Function]}
      onCalendarClose={[Function]}
      onCalendarOpen={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      preventOpenOnFocus={false}
      previousMonthButtonLabel="Previous Month"
      previousYearButtonLabel="Previous Year"
      readOnly={false}
      renderDayContents={[Function]}
      selected={1970-01-01T00:00:00.000Z}
      shouldCloseOnSelect={true}
      showFourColumnMonthYearPicker={false}
      showFullMonthYearPicker={false}
      showMonthYearPicker={false}
      showPopperArrow={true}
      showPreviousMonths={false}
      showQuarterYearPicker={false}
      showTimeInput={false}
      showTimeSelect={true}
      showTwoColumnMonthYearPicker={false}
      showYearPicker={false}
      strictParsing={false}
      timeCaption="Time"
      timeFormat="HH:mm"
      timeInputLabel="Time"
      timeIntervals={30}
      withPortal={false}
      yearItemNumber={12}
    />
    <button>
      Save Order
    </button>
  </form>
  <div>
    <p>
      customerName: 
      Ashath
    </p>
    <p>
      phoneNumber: 
      0765822562
    </p>
    <p>
      address: 
      Mamangam
    </p>
    <p>
      description: 
      less oil
    </p>
    <p>
      orderEndTime:  
      Thu Jan 01 1970 05:50:00 GMT+0530
    </p>
    <p>
      status: 
      confirmed
    </p>
    <p>
      kotSatus: 
      passed
    </p>
    <p>
      billStatus: 
      printed
    </p>
    <p>
      createdAt: 
      Thu Jan 01 1970 05:30:00 GMT+0530
    </p>
    <p>
      foods: 
      [object Object],[object Object]
    </p>
    <p>
      isModaOpen: 
      false
    </p>
    <p>
      deliverers: 
      Ashath
    </p>
  </div>
</div>
`;

exports[`should render error for invalid form submission 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Add Food
    </button>
    <Connect(OrderModal)
      isModalOpen={false}
      onSingleOrderChange={[Function]}
      onVisibleChange={[Function]}
      singleTypeOrders={Array []}
    />
    <input
      onChange={[Function]}
      placeholder="Customer Name"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Phone Number"
      type="number"
      value={0}
    />
    <input
      onChange={[Function]}
      placeholder="Customer Address"
      type="text"
      value=""
    />
    <textarea
      autoFocus={true}
      onChange={[Function]}
      placeholder="Add a discription for your order (optional)"
      value=""
    />
    <div>
      <h3>
        :::::::::::::
        0
        :::::::::::::
      </h3>
    </div>
    <select
      onChange={[Function]}
      value="confirmed"
    >
      <option
        value="confirmed"
      >
        Confirmed
      </option>
      <option
        value="cancelled"
      >
        Cancelled
      </option>
      <option
        value="kitchen"
      >
        Kitchen
      </option>
      <option
        value="table"
      >
        Table
      </option>
      <option
        value="on delivery"
      >
        On Delivery
      </option>
      <option
        value="recieved"
      >
        Recieved
      </option>
    </select>
    <select
      onChange={[Function]}
      value=""
    >
      <option
        selected={true}
        value=""
      >
        Select a Deliverer
      </option>
      <option
        value="defaultId"
      >
        default
      </option>
    </select>
    <select
      disabled={true}
      onChange={[Function]}
      value="not"
    >
      <option
        value="not"
      >
        Not Passed
      </option>
      <option
        value="passed"
      >
        Passed
      </option>
    </select>
    <select
      disabled={true}
      onChange={[Function]}
      value="not"
    >
      <option
        value="not"
      >
        Not Printed
      </option>
      <option
        value="printed"
      >
        Printed
      </option>
    </select>
    <select
      onChange={[Function]}
      value="20"
    >
      <option
        value="20"
      >
        20 minutes
      </option>
      <option
        value="30"
      >
        30 minutes
      </option>
      <option
        value="40"
      >
        40 minutes
      </option>
      <option
        value="50"
      >
        50 minutes
      </option>
      <option
        value="60"
      >
        1 hour
      </option>
      <option
        value="70"
      >
        1 hour 10 minutes
      </option>
      <option
        value="80"
      >
        1 hour 20 minute
      </option>
      <option
        value="90"
      >
        1 and 1\\2 hour 
      </option>
    </select>
    <r
      allowSameDay={false}
      calendarStartDay={0}
      customTimeInput={null}
      dateFormat="MMMM d, yyyy h:mm aa"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      enableTabLoop={true}
      excludeScrollbar={true}
      focusSelectedMonth={false}
      monthsShown={1}
      nextMonthButtonLabel="Next Month"
      nextYearButtonLabel="Next Year"
      onBlur={[Function]}
      onCalendarClose={[Function]}
      onCalendarOpen={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      preventOpenOnFocus={false}
      previousMonthButtonLabel="Previous Month"
      previousYearButtonLabel="Previous Year"
      readOnly={false}
      renderDayContents={[Function]}
      selected={1970-01-01T00:00:00.000Z}
      shouldCloseOnSelect={true}
      showFourColumnMonthYearPicker={false}
      showFullMonthYearPicker={false}
      showMonthYearPicker={false}
      showPopperArrow={true}
      showPreviousMonths={false}
      showQuarterYearPicker={false}
      showTimeInput={false}
      showTimeSelect={true}
      showTwoColumnMonthYearPicker={false}
      showYearPicker={false}
      strictParsing={false}
      timeCaption="Time"
      timeFormat="HH:mm"
      timeInputLabel="Time"
      timeIntervals={30}
      withPortal={false}
      yearItemNumber={12}
    />
    <button>
      Save Order
    </button>
  </form>
  <div>
    <p>
      customerName: 
    </p>
    <p>
      phoneNumber: 
      0
    </p>
    <p>
      address: 
    </p>
    <p>
      description: 
    </p>
    <p>
      orderEndTime:  
      Thu Jan 01 1970 05:50:00 GMT+0530
    </p>
    <p>
      status: 
      confirmed
    </p>
    <p>
      kotSatus: 
      not
    </p>
    <p>
      billStatus: 
      not
    </p>
    <p>
      createdAt: 
      Thu Jan 01 1970 05:30:00 GMT+0530
    </p>
    <p>
      foods: 
    </p>
    <p>
      isModaOpen: 
      false
    </p>
    <p>
      deliverers: 
    </p>
  </div>
</div>
`;

exports[`should render error for invalid form submission 2`] = `
<div>
  <p>
    CUSTOMER NAME, PHONENUMBER, ADDRESS , ORDERDURATIONS are ORDER PLACINGS are mandatoroy to fill
  </p>
  <form
    onSubmit={[Function]}
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Add Food
    </button>
    <Connect(OrderModal)
      isModalOpen={false}
      onSingleOrderChange={[Function]}
      onVisibleChange={[Function]}
      singleTypeOrders={Array []}
    />
    <input
      onChange={[Function]}
      placeholder="Customer Name"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Phone Number"
      type="number"
      value={0}
    />
    <input
      onChange={[Function]}
      placeholder="Customer Address"
      type="text"
      value=""
    />
    <textarea
      autoFocus={true}
      onChange={[Function]}
      placeholder="Add a discription for your order (optional)"
      value=""
    />
    <div>
      <h3>
        :::::::::::::
        0
        :::::::::::::
      </h3>
    </div>
    <select
      onChange={[Function]}
      value="confirmed"
    >
      <option
        value="confirmed"
      >
        Confirmed
      </option>
      <option
        value="cancelled"
      >
        Cancelled
      </option>
      <option
        value="kitchen"
      >
        Kitchen
      </option>
      <option
        value="table"
      >
        Table
      </option>
      <option
        value="on delivery"
      >
        On Delivery
      </option>
      <option
        value="recieved"
      >
        Recieved
      </option>
    </select>
    <select
      onChange={[Function]}
      value=""
    >
      <option
        selected={true}
        value=""
      >
        Select a Deliverer
      </option>
      <option
        value="defaultId"
      >
        default
      </option>
    </select>
    <select
      disabled={true}
      onChange={[Function]}
      value="not"
    >
      <option
        value="not"
      >
        Not Passed
      </option>
      <option
        value="passed"
      >
        Passed
      </option>
    </select>
    <select
      disabled={true}
      onChange={[Function]}
      value="not"
    >
      <option
        value="not"
      >
        Not Printed
      </option>
      <option
        value="printed"
      >
        Printed
      </option>
    </select>
    <select
      onChange={[Function]}
      value="20"
    >
      <option
        value="20"
      >
        20 minutes
      </option>
      <option
        value="30"
      >
        30 minutes
      </option>
      <option
        value="40"
      >
        40 minutes
      </option>
      <option
        value="50"
      >
        50 minutes
      </option>
      <option
        value="60"
      >
        1 hour
      </option>
      <option
        value="70"
      >
        1 hour 10 minutes
      </option>
      <option
        value="80"
      >
        1 hour 20 minute
      </option>
      <option
        value="90"
      >
        1 and 1\\2 hour 
      </option>
    </select>
    <r
      allowSameDay={false}
      calendarStartDay={0}
      customTimeInput={null}
      dateFormat="MMMM d, yyyy h:mm aa"
      dateFormatCalendar="LLLL yyyy"
      disabled={false}
      disabledKeyboardNavigation={false}
      dropdownMode="scroll"
      enableTabLoop={true}
      excludeScrollbar={true}
      focusSelectedMonth={false}
      monthsShown={1}
      nextMonthButtonLabel="Next Month"
      nextYearButtonLabel="Next Year"
      onBlur={[Function]}
      onCalendarClose={[Function]}
      onCalendarOpen={[Function]}
      onChange={[Function]}
      onClickOutside={[Function]}
      onFocus={[Function]}
      onInputClick={[Function]}
      onInputError={[Function]}
      onKeyDown={[Function]}
      onMonthChange={[Function]}
      onSelect={[Function]}
      onYearChange={[Function]}
      preventOpenOnFocus={false}
      previousMonthButtonLabel="Previous Month"
      previousYearButtonLabel="Previous Year"
      readOnly={false}
      renderDayContents={[Function]}
      selected={1970-01-01T00:00:00.000Z}
      shouldCloseOnSelect={true}
      showFourColumnMonthYearPicker={false}
      showFullMonthYearPicker={false}
      showMonthYearPicker={false}
      showPopperArrow={true}
      showPreviousMonths={false}
      showQuarterYearPicker={false}
      showTimeInput={false}
      showTimeSelect={true}
      showTwoColumnMonthYearPicker={false}
      showYearPicker={false}
      strictParsing={false}
      timeCaption="Time"
      timeFormat="HH:mm"
      timeInputLabel="Time"
      timeIntervals={30}
      withPortal={false}
      yearItemNumber={12}
    />
    <button>
      Save Order
    </button>
  </form>
  <div>
    <p>
      customerName: 
    </p>
    <p>
      phoneNumber: 
      0
    </p>
    <p>
      address: 
    </p>
    <p>
      description: 
    </p>
    <p>
      orderEndTime:  
      Thu Jan 01 1970 05:50:00 GMT+0530
    </p>
    <p>
      status: 
      confirmed
    </p>
    <p>
      kotSatus: 
      not
    </p>
    <p>
      billStatus: 
      not
    </p>
    <p>
      createdAt: 
      Thu Jan 01 1970 05:30:00 GMT+0530
    </p>
    <p>
      foods: 
    </p>
    <p>
      isModaOpen: 
      false
    </p>
    <p>
      deliverers: 
    </p>
  </div>
</div>
`;
